ember-client/src/main/scala/org/http4s/ember/client/internal/ClientHelpers.scala

```diff
@@ -100,19 +100,20 @@ private[client] object ClientHelpers {
         .compile
         .drain
 
-    def writeRead(req: Request[F]): F[Response[F]] =
-      writeRequestToSocket(req, requestKeySocket.socket, durationToFinite(idleTimeout)) >> {
-        Parser.Response
-          .parser(maxResponseHeaderSize, durationToFinite(timeout))(
-            requestKeySocket.socket.reads(chunkSize, durationToFinite(idleTimeout))
-          )
-          .map(_._1)
-      }
+    def writeRead(req: Request[F]): F[(Response[F], F[Option[Array[Byte]]])] =
+      writeRequestToSocket(req, connection.keySocket.socket, durationToFinite(idleTimeout)) >>
+        connection.nextBytes.getAndSet(Array.emptyByteArray).flatMap { head =>
+          Parser.Response
+            .parser(maxResponseHeaderSize, durationToFinite(timeout))(
+              head,
+              connection.keySocket.socket.read(chunkSize, durationToFinite(idleTimeout))
+            )
+        }
 
     for {
       processedReq <- preprocessRequest(request, userAgent)
-      resp <- writeRead(processedReq)
-    } yield postProcessResponse(processedReq, resp, reuseable)
+      (resp, drain) <- writeRead(processedReq)
```

Comment:
This won't compile in Scala 3, but we can deal with it there.

Reply:
It disappeared anyways